### PR TITLE
chore: fix utils/utils.go / repoDir

### PR DESCRIPTION
**Pull request by atsushifx**
- first, reference $XDG_CACHE_HOME directory to All OS's cache directory
  - fix: change save directory gibo to gibo/gitignore-boilerplates
  - fix: compare string change == to string.equalFold

### CODE REVIEW

File: utils/utils.go [hunk 1]
```diff
@@ -14,11 +14,17 @@ import (
 )
 
 func RepoDir() string {
-	cacheDir, err := os.UserCacheDir()
-	if err != nil {
-		log.Fatalln("gibo can't determine your user cache directory. Please file an issue at https://github.com/simonwhitaker/gibo/issues")
+	xdgCacheDir := os.Getenv("XDG_CACHE_HOME")
```
Comment (simonwhitaker) [inline utils/utils.go:17]: What's your intention here? `os.UserCacheDir()` will respect `XDG_CACHE_HOME` on Unix systems. I suppose people on macOS or Windows might set the `XDG_CACHE_HOME` environment variable, but I don't think it makes sense to try to honour non-standard configs; the logic here would potentially get very complicated, for little benefit.

File: utils/utils.go [hunk 2]
```diff
@@ -56,7 +62,7 @@ func pathForBoilerplate(name string) (string, error) {
 	filename := name + ".gitignore"
 	var result string = ""
 	filepath.WalkDir(RepoDir(), func(path string, d fs.DirEntry, err error) error {
-		if strings.ToLower(filepath.Base(path)) == strings.ToLower(filename) {
+		if strings.EqualFold(filepath.Base(path), filename) {
```
Comment (simonwhitaker) [inline utils/utils.go:65]: Nice, I didn't know about `strings.EqualFold`. Looks useful!

**Review by simonwhitaker — CHANGES_REQUESTED** [review]
Please remove the code for explicitly checking `XDG_CACHE_HOME`; we can just rely on `os.UserCacheDir()` here.

File: utils/utils.go (hunk 1 above)
Comment (atsushifx) [reply utils/utils.go]: I understood. New commit:[13959a970eda9e4c61afe6105f176c01f9d14ff5](https://github.com/simonwhitaker/gibo/pull/63/commits/13959a970eda9e4c61afe6105f176c01f9d14ff5) remove these codes relied XDG_CACHE_HOME.
please check them.

**Review by simonwhitaker — APPROVED** [review]

**Pull request merged**